src/panels/lovelace/cards/hui-map-card.ts

```diff
@@ -329,6 +386,63 @@ class HuiMapCard extends LitElement implements LovelaceCard {
       }
     }
 
+    // DRAW history
+    if (this._config!.hours_to_show && this._history) {
+      for (const entityStates of this._history) {
+        if (entityStates?.length <= 1) {
+          continue;
+        }
+        const entityId = entityStates[0].entity_id;
+        const entity: EntityConfig = this._configEntities!.filter(
+          (x) => x.entity === entityId
+        )[0];
+
+        // filter location data from states and remove all invalid locations
+        const path = entityStates.reduce(
+          (accumulator: LatLngTuple[], state) => {
+            const latitude = state.attributes.latitude;
+            const longitude = state.attributes.longitude;
+            if (latitude && longitude) {
+              accumulator.push([latitude, longitude] as LatLngTuple);
+            }
+            return accumulator;
+          },
+          []
+        ) as LatLngTuple[];
+
+        // DRAW HISTORY
+        for (
+          let markerIndex = 0;
+          markerIndex < path.length - 1;
+          markerIndex++
+        ) {
+          const opacityStep = 1 / path.length;
+          const opacity = 2 * opacityStep + markerIndex * opacityStep;
```

Comment:
If we simplify the above equation, you get:
```suggestion
          const opacity = (markerIndex + 2) * opacityStep;
```
This results in an opacity above 1?

Reply:
Actually not, because the loop condition is `markerIndex < path.length - 1 `. (I do not draw the dot underneath the current location). 
Same code here:  https://js.do/code/423192

But you are right, somehow I missed to start at 20% opacity. 
New code is be: https://js.do/code/423195